app/models/project.rb

```diff
@@ -70,6 +70,12 @@ def is_not_final?
     self.title.blank? || self.language.blank? || self.description.blank? || self.github_link.blank?
   end
 
+  def tasks_completed_for_week?(week_number)
+    tasks_for_week = self.tasks.where(week: week_number)
```

Comment:
you don't need to use `self` just `tasks` is enough. Self is mostly used when we want to assign new value to record

Reply:
good point.